compiler-core/src/build/package_loader.rs

```diff
@@ -157,11 +157,18 @@ where
 
                 // A cached module with no stale dependencies can be used as-is
                 // and does not need to be recompiled.
-                Input::Cached(info) => {
-                    tracing::debug!(module = %info.name, "module_to_load_from_cache");
-                    let module = self.load_cached_module(info)?;
-                    loaded.cached.push(module);
-                }
+                Input::Cached(info) => match self.mode {
+                    Mode::PackageInterface => {
```

Comment:
This always compiles modules, even if they have already been compiled.
This isn't the best experience as it repeats already done work, it would be better to build the interface from the cache I think.

Reply:
Fixed in commit 11c1d18!